### ISSUE TITLE
Better block explorer

### ISSUE THREAD

**Issue by acolytec3** [ethereumjs/ultralight#161]
We should either fork the current `browser-client` or else make the block explorer a more standalone component that is easy to use in terms of searching for and displaying blocks

Ideas for a better block explorer
- Remove all the Portal Network specific buttons
- Replace the `sendFindContent` button and input with a single "Enter Blockhash" input and "Search" button
- Hardcode a set of bootnodes to connect to (this is dependent on there being a public test network with bootnodes that serve actual valid content)
- Make the Parent Block hash searchable so if you click on it, a `sendFindContent` request is sent to the network to retrieve the corresponding block and display it in the block explorer

**Comment by holgerd77** [comment]
Would it also be a way to use an open source block explorer and just provide a Portal Network connector to feed with the data? Or has this kind of block explorer such specific needs that you would want to continue to develop top to bottom?

**Comment by acolytec3** [comment]
Possibly down the line. I looked at Otterscan (for example) and it's pretty tightly integrated with Erigon. I can look at BlockScout and see how it works as that might be an interesting place to look at working in an integration. This issue attend directly from a conversation @ScottyPoi and I had yesterday about getting something a little more user friendly ready for Dev connect.

Do you know of any other open source block explorers that have any sort of use/general community support?

**Comment by holgerd77** [comment]
Sorry, not too much insight here, I like BlockScout (without digging deeper) being also deployed for the Kiln Merge testnet: https://explorer.kiln.themerge.dev/

**Comment by acolytec3** [comment]
No worries.  I took a look at the Blockscout repo and I don't think it's feasible to integrate with at this point.  They need a full archive node and presumably support for all the normal `eth_...` JSONRPC API methods.  I think the biggest challenge for a real, fully functional block explorer is the portal network only supports `eth_getBlockByHash` at present so the only way for us to really index blocks for a block explorer would be to maintain an index of all block numbers to block hashes locally and then start retrieving the blocks one by one from the network.  So, for better or for worse, our block explorer is really just for retrieving block details where you know the hash of the block you're looking for.

**Comment by holgerd77** [comment]
Yeah, sure, if it's more practical right now (or/and in the future as well) very much go for it. 🙂 Was just an ad-hoc question.
